### DISCUSSION TITLE
Display dates and times

### DISCUSSION THREAD

**Discussion by tomyems** [UKHomeOffice/design-system#386] [Ideas]
A proof of concept has been developed that can display dates and times in a format that is easy for a human to read, as opposed to the ISO format in which is it stored:

- Date is formatted from an ISO datetime string to meet GOV style guidelines
- Date can be displayed in a configured format
- Date can be displayed with or without time in a configured order

There is some technical detail https://github.com/UKHomeOffice/design-system/pull/350 alongside guidance about when to use the formats, with the current version available https://626b9e86a2864b53b69d5476--hods.netlify.app/.

**Comment by tomyems** [comment]
From the monthly design crit:

- place all the guidance on one page, as just different variations on the component
- provide the next steps in the contribute on how to contribute to the component
- make consistent with the rest of the components about where the main example is
- reduce the duplication in the time format section
- the timeline component will need updating so that the date and time are consistent with this method of displaying dates and times

One internal system did use dates and times combined in a timeline activity log and in file uploads. It displayed times in the 12 hour clock.

As a result of this feedback the guidance has been iterated, with the current version being: https://62729855d02ed2165eab2204--hods.netlify.app/components?name=Date%20and%20time%20display

**Comment by tomyems** [comment]
Following additional feedback that:

- there were too many examples which was confusing
- the heading 1 didn't match the text in the menu on the left hand side

See the current version: https://62751cec792e88117276676c--hods.netlify.app/components?name=Date%20and%20time%20display

**Comment by tomyems** [comment]
This is now being delivered through https://github.com/UKHomeOffice/design-system/pull/350.

**Comment by tomyems** [comment]
Published: https://design.homeoffice.gov.uk/components?name=Date-time

If you have any suggestions or improvements add them to this discussion. Including if you have any insights on:
- when to use the 24-hour clock
- displaying time zones
- when to display the date before the time, or the time before the date

**Comment by tomyems** [comment]
Suggestion from @Sarah-Lombardi (https://github.com/UKHomeOffice/design-system/issues/554) to change the label in the menu and the heading on the component page (https://design.homeoffice.gov.uk/components?name=Date-time) to 'Date and time'

**Comment by Sarah-Lombardi** [comment]
I reviewed this as I think the content of the 2 'date' components duplicate each other. I have proposed a slightly different way of doing it, merging the two and removing duplication. See attached. I propose keeping 'Time' mostly as it is, maybe with some minor edits to differentiate it from this one.

@daniel-ac-martin I think you were involved in this before and may have some background about why it was separated? Perhaps something to discuss at our working group.

**Comment by Sarah-Lombardi** [comment]
File won't load so copying below:

# Dates

Use this code component to display dates that are easy to read. 

You can do this by adding in a ‘datetime’ tag that includes a:

- standard date, such as 2022-07-20
- the full [ISO format](https://en.wikipedia.org/wiki/ISO_8601) for date and time, such as 2022-07-21T22:37:34.570Z

The content follows the [GOV.UK time guidance](https://www.gov.uk/guidance/style-guide/a-to-z#times) and the [GOV.UK date guidance](https://www.gov.uk/guidance/style-guide/a-to-z#dates).

## Date

Add component here

## Date and time

Add component here

This includes the [full ISO formatted date-time](https://en.wikipedia.org/wiki/ISO_8601). You can display just the time, as calculated from the timezone.

Use the 12-hour clock – for example 5:30pm – unless research shows that the 24-hour clock is easier. This is sometimes true in our internal services. 

If evidence shows time is more important than date, you can change the order. The ISO code remains the same.

Add component here

## Accessibility

If your service uses these components, let us know if you have tested how accessible they are.

## Help improve this component

This component needs improving. We need evidence about:

- when to use the 24-hour clock
- displaying time zones
- when to display the date before the time, or the time before the date
- how to write for this component
- which services are using this component

To contribute, add your thoughts and research findings to our [GitHub discussion](https://github.com/UKHomeOffice/design-system/discussions/386), or follow our [contribute guidance](https://design.homeoffice.gov.uk/contribute)

**Comment by daniel-ac-martin** [comment]
Hi @Sarah-Lombardi.

I think there is a subtle distinction between the two. We have three components:
- **DateDisplay:** For displaying a date only.
- **Time:** For displaying a time only.
- **DateTime:** For displaying a specific time on a specific date. (i.e. The above together.)

You might have a timestamp (full date-time), but only care about surfacing the date, because the time doesn't matter. In that case you would just use DateDisplay rather than DateTime.

We could insist that the programmer removes the date-time parts that they don't want, but this unnecessarily pushes work onto consumers.

It would be possible to combine the components into one by adding more toggles to control the formatting, but that seems less clean to me.

Overall, I'd suggest keeping them separate and maybe tweaking the content on the pages, perhaps to refer to one another.

**Reply by Sarah-Lombardi** [reply]
Ah I see. So they all use the date-time tag but they use different classes if they just want date, or date and time, or just time? I don't think that is clear so we could draw that out.

I also still think that the distinction is very minor and we don't tend to split our components up that way across gov design systems - for instance the radio component includes conditional reveals and inclusion of a 'none' option. 

I'll bring it to todays working group for discussion anyway.
